### ISSUE TITLE
what's the meaning and relationship of TaskList, SubmitList, MemoryList and MemoryList in the loadable file??

### ISSUE THREAD

**Issue by bufans** [nvdla/sw#192]
m_task_entries   = loadable->getTaskListEntries();
    m_submit_entries = loadable->getSubmitListEntries();
    m_memory_entries = loadable->getMemoryListEntries();
    m_address_entries = loadable->getAddressListEntries();
    m_tensor_desc_entries = loadable->getTensorDescListEntries();
    m_reloc_entries = loadable->getRelocEntries();

**Comment by tymcauley** [comment]
These all refer to variables extracted from an [NVDLA Loadable](http://nvdla.org/sw/runtime_environment.html) (which describes a Neural Network workload for the NVDLA). You can see the format of that binary file (called a [FlatBuffer](http://google.github.io/flatbuffers/)) in this file: `umd/core/src/common/loadable.fbs`

I actually have a related question: What is the purpose of the `RelocListEntry` in that file? It doesn't appear to have much use in the UMD, but I see those entries filled out for some networks, such as the precompiled LeNet FlatBuffer (`regression/flatbufs/kmd/NN/NN_L0_0_fbuf`).

**Comment by bufans** [comment]
Is one task corresponding to one neural network ? 
Can a loadable file contain multi neural network ?

**Comment by tymcauley** [comment]
I believe a single Loadable only describes a single neural network. However, a single Loadable can have multiple Tasks. For example, a simple classifier network, which usually ends with a Softmax layer, has two tasks: one runs the body of the network on the NVDLA, and the other task runs the final Softmax layer on the CPU (since the NVDLA doesn't implement that in hardware). In the UMD, this CPU implementation of certain layers is referred to as the "Emulator".

**Comment by bufans** [comment]
OK, thanks, this helps a lot.

**Comment by bufans** [comment]
> These all refer to variables extracted from an [NVDLA Loadable](http://nvdla.org/sw/runtime_environment.html) (which describes a Neural Network workload for the NVDLA). You can see the format of that binary file (called a [FlatBuffer](http://google.github.io/flatbuffers/)) in this file: `umd/core/src/common/loadable.fbs`
> 
> I actually have a related question: What is the purpose of the `RelocListEntry` in that file? It doesn't appear to have much use in the UMD, but I see those entries filled out for some networks, such as the precompiled LeNet FlatBuffer (`regression/flatbufs/kmd/NN/NN_L0_0_fbuf`).

I found the ReLocatioEntry is used in runtime::rewriteStride, which is used in setupInputTensor.  Maybe the ReLocationEntry is used to help change  address list, since the input cube/tensor is not contained in the loadable file ?
